setup.py

```diff
@@ -1116,6 +1116,8 @@ def main():
         "jinja2",
         "fsspec",
     ]
+    if IS_WINDOWS:
+        install_requires.append("mkl>=2021.1.1,<=2021.4.0")
```

Comment:
I was thinking first to have this implementation since previously we were using MKL 2020. See if everything works well and then in the future will be easier to just change the version to 2022.

Reply:
@mantaionut do you still want to update it to 2022? Also, are you sure that we are not double packing mkl into the wheel, but keeping dependency at the same time?

Reply:
@malfet the binaries of mkl are not copied in the wheel, only in libtorch they are copied. I created a draft PR https://github.com/pytorch/pytorch/pull/118200 for updating to 2022. However I see for conda it might not be possible since numpy 1.19  -> mkl[version='>=2019.4,<2021.0a0|>=2021.4.0,<2022.0a0|>=2023.1.0,<2024.0a0']. 
So based on this we could update to 2022 however on conda we should use 2023 instead. Let me know if you think this will work for you.